wd.html

```diff
@@ -444,6 +444,276 @@ <h3>Get Title</h3>
   <code class=cs>this.driver.Title;</code>
  </pre>
 
+<h2>Windows and tabs</h2>
+<p>WebDriver doesn't make the distinction between windows and tabs.  If
+ your site opens a new tab or window, Selenium will let you work with it
+ using a window handle.  Each window has a unique identifier which remains
+ persistent in a single session. You can get the window handle of the
+ current window by using:</p>
+<pre>
+ <code class=java>driver.getWindowHandle();</code>
+ <code class=python>driver.current_window_handle</code>
+ <code class=cs>driver.CurrentWindowHandle</code>
+ <code class=ruby>driver.window_handle</code>
+ <code class=javascript>await driver.getWindowHandle();</code>
+</pre>
+<h3>Switching windows or tabs</h3>
+<p>Clicking a link which opens in a <a href="https://seleniumhq.github.io/docs/" target="_blank">
+ new window</a> will visible focus the new window or tab on screen, but
+ WebDriver will not know which window the Operating System considers
+ active.  To work with the new window you will need to switch to it.  If
+ you have only two tabs or windows open, and you know which window you
+ start with, by the process of elimination you can loop over both windows
+ or tabs that WebDriver can see, and switch to the one which is not the
+ original.</p>
+<pre>
+ <code class=java>//Store the ID of the original window
+String originalWindow = driver.getWindowHandle();
+
+//Check we don't have other windows open already
+assert driver.getWindowHandles().size() == 1;
+
+//Click the link which opens in a new window
+driver.findElement(By.linkText("new window")).click();
+
+//Wait for the new window or tab
+wait.until(numberOfWindowsToBe(2));
+
+//Loop through until we find a new window handle
+for (String windowHandle : driver.getWindowHandles()) {
+    if(!originalWindow.contentEquals(windowHandle)) {
+        driver.switchTo().window(windowHandle);
+        break;
+    }
+}
+
+//Wait for the new tab to finish loading content
+wait.until(titleIs("Selenium documentation"));</code>
+ <code class=python>#Store the ID of the original window
+original_window = driver.current_window_handle
+
+#Check we don't have other windows open already
+assert len(driver.window_handles) == 1
+
+#Click the link which opens in a new window
+driver.find_element(By.LINK_TEXT, "new window").click()
+
+#Wait for the new window or tab
+wait.until(EC.number_of_windows_to_be(2))
```

Comment:
Might want to either show the import and alias, or use full `ExpectedConditions` class. Thinking of new beginners. Also, similar with `wait`. May want to show how that's instantiated `WebDriverWait(driver, timeout)`.

Alternatively, you could link to the `waits` section in the leading text

Reply:
Spelling out how to do waits properly is probably a good idea even if it leads to repetition.

Reply:
How about:
```from selenium import webdriver
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

# Start the driver
with webdriver.Firefox() as driver:
    # Setup wait for later
    wait = WebDriverWait(driver, 10)
    
    # Store the ID of the original window
    original_window = driver.current_window_handle
    
    # Check we don't have other windows open already
    assert len(driver.window_handles) == 1
    
    # Click the link which opens in a new window
    driver.find_element_by_link_text("new window").click()
    
    # Wait for the new window or tab
    wait.until(EC.number_of_windows_to_be(2))
    
    # Loop through until we find a new window handle
    for window_handle in driver.window_handles:
        if window_handle != original_window:
            driver.switch_to.window(window_handle)
            break
    
    # Wait for the new tab to finish loading content
    wait.until(EC.title_is("Selenium documentation"))
```

Reply:
Sure, looks good